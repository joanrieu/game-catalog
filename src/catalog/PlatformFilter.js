import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Wrapper = styled.nav`
  padding: 16px;
`;

const Platform = styled.button`
  border: none;
  background: transparent;
  font: inherit;
  color: ${props => (props.isSelected ? "#FE6050" : "inherit")}
  outline-color: #FE6050;
  text-transform: uppercase;

  &:not(:last-child) {
    margin-right: 28px;
  }
`;

export default function PlatformFilter({
  platforms,
  selectedPlatform,
  selectPlatform
}) {
  return (
    <Wrapper>
      <Platform onClick={() => selectPlatform()} isSelected={!selectedPlatform}>
        All
      </Platform>
      {platforms.map(platform => (
        <Platform
          key={platform}
          onClick={() => selectPlatform(platform)}
          isSelected={platform === selectedPlatform}
        >
          {platform}
        </Platform>
      ))}
    </Wrapper>
  );
}

PlatformFilter.propTypes = {
  platforms: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  selectedPlatform: PropTypes.string,
  selectPlatform: PropTypes.func.isRequired
};
