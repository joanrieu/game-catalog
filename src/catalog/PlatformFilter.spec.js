import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import PlatformFilter from "./PlatformFilter";

describe("PlatformFilter", () => {
  it("has a list of platforms", () => {
    const cmp = renderer.create(
      <PlatformFilter
        platforms={["PC", "PS4", "Xbox"]}
        selectedPlatform="PS4"
        selectPlatform={() => {}}
      />
    );
    expect(cmp).toMatchSnapshot();
  });
  it("can change the selected platform", () => {
    const selectMock = jest.fn();
    const cmp = renderer.create(
      <PlatformFilter
        platforms={["PC", "PS4", "Xbox"]}
        selectedPlatform="PS4"
        selectPlatform={selectMock}
      />
    );
    const buttons = cmp.root.findAllByType("button");
    expect(buttons.length).toBe(4);
    buttons[3].props.onClick();
    expect(selectMock).toHaveBeenCalledWith("Xbox");
  });
  it("can clear the selected platform", () => {
    const selectMock = jest.fn();
    const cmp = renderer.create(
      <PlatformFilter
        platforms={["PC", "PS4", "Xbox"]}
        selectPlatform={selectMock}
      />
    );
    const buttons = cmp.root.findAllByType("button");
    expect(buttons.length).toBe(4);
    buttons[0].props.onClick();
    expect(selectMock).toHaveBeenCalledWith();
  });
});
