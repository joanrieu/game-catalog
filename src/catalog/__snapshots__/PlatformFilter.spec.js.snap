// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformFilter has a list of platforms 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdVaJa eVCtOy"
  >
    <button
      class="sc-bwzfXH cIfOCo"
    >
      All
    </button>
    <button
      class="sc-bwzfXH cIfOCo"
    >
      PC
    </button>
    <button
      class="sc-bwzfXH hsBwin"
    >
      PS4
    </button>
    <button
      class="sc-bwzfXH cIfOCo"
    >
      Xbox
    </button>
  </nav>
</DocumentFragment>
`;
